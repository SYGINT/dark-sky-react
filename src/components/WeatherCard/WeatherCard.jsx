import React from "react";

import WeatherIcon from "../WeatherIcon";
import WeatherDetail from "../WeatherDetail";

import "./WeatherCard.scss";

function WeatherCard({ weather }) {
  const { currently, hourly } = weather;
  const { icon, summary: currentSummary, temperature } = currently;
  const { summary: hourlySummary } = hourly;

  const roundedTemperature = Math.round(temperature);

  return (
    <div className="ui comments">
      <div className="comment">
        <WeatherIcon icon={icon} />

        <div className="content summary">
          <div className="title">
            {roundedTemperature}&deg; {currentSummary}.
          </div>
          <div className="details">
            <WeatherDetail name="Feels like" value="x" />
            <WeatherDetail name="Low" value="x" />
            <WeatherDetail name="High" value="x" />
          </div>
          <div>{hourlySummary}</div>
        </div>
      </div>
    </div>
  );
}

export default WeatherCard;
