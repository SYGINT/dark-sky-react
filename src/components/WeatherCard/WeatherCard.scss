// override Semantic
.comments {
  display: inline-block;
  text-align: left;
}

.weather-long-summary {
  font-size: 28px;
}

.weather-main {
  font-size: 32px;
  font-weight: 600;
  line-height: 1.3em;
}

.content.summary {
  padding-top: 5px;
}
