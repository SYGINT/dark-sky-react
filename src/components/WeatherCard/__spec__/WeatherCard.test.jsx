import React from "react";
import { shallow } from "enzyme";

import WeatherCard from "../";

it("renders without crashing", () => {
  const mockWeather = {
    currently: {
      icon: "clear-night",
      summary: "Clear",
      temperature: 55.71
    },
    hourly: {
      summary: "Partly cloudy starting tomorrow morning."
    },
    timezone: "America/Los_Angeles"
  };
  const wrapper = shallow(<WeatherCard weather={mockWeather} />);
  expect(wrapper).toMatchSnapshot();
});
