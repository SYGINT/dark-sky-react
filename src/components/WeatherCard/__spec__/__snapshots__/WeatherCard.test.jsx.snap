// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="ui comments"
>
  <div
    className="comment"
  >
    <WeatherIcon
      icon="clear-night"
    />
    <div
      className="content summary"
    >
      <div
        className="title"
      >
        56
        Â° 
        Clear
        .
      </div>
      <div
        className="details"
      >
        <WeatherDetail
          name="Feels like"
          value="x"
        />
        <WeatherDetail
          name="Low"
          value="x"
        />
        <WeatherDetail
          name="High"
          value="x"
        />
      </div>
      <div>
        Partly cloudy starting tomorrow morning.
      </div>
    </div>
  </div>
</div>
`;
